### PR TITLE
Make it easier to use memchecks in HLE

### PR DESCRIPTION
**Pull request by unknownbrackets**
Really, it makes sense for `Memory::Memcpy()` to do this directly.

Anyway, I used a separate header since I brought in other headers and stuff.

-[Unknown]

### CODE REVIEW

File: Core/MemMapHelpers.h [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		CBreakPoints::ExecMemCheck(from_address, false, len, currentMIPS->pc);
+#endif
+	}
+	// if not, GetPointer will log.
+}
+
+inline void Memcpy(const u32 to_address, const u32 from_address, const u32 len)
+{
+#ifndef MOBILE_DEVICE
+	CBreakPoints::ExecMemCheck(to_address, true, len, currentMIPS->pc);
```
Comment (hrydgard) [inline Core/MemMapHelpers.h]: Hm, it's not entirely consistent across the functions whether we do the memcpy or the memcheck first. I don't really care either way but they should all be the same.

Comment (unknownbrackets) [inline Core/MemMapHelpers.h]: Yeah, that makes sense.

-[Unknown]

File: Core/MemMapHelpers.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+#ifndef MOBILE_DEVICE
+		CBreakPoints::ExecMemCheck(from_address, false, len, currentMIPS->pc);
+#endif
+	}
+	// if not, GetPointer will log.
+}
+
+inline void Memcpy(const u32 to_address, const u32 from_address, const u32 len)
+{
+	Memcpy(GetPointer(to_address), from_address, len);
```
Comment (hrydgard) [inline Core/MemMapHelpers.h]: Hm, you call Memcpy (with a pointer so no self-recursion) instead of memcpy. Won't this end up doing memchecks twice?

Comment (unknownbrackets) [inline Core/MemMapHelpers.h]: This is intentional, There are two memchecks to do here (the from range and the to range), and so it's doing one in the outer Memcpy and the other in the inner Memcpy.

-[Unknown]

Comment (hrydgard) [inline Core/MemMapHelpers.h]: Oh, right.

**Pull request merged**